src/libappimage/type2.c

```diff
@@ -101,11 +105,11 @@ bool appimage_type2_resolve_symlink(sqfs* fs, sqfs_inode* inode) {
         return false;
     }
 
-    return true;
+    return appimage_type2_resolve_symlink(fs, inode, hops_remaining - 1);
 }
 
 bool appimage_type2_extract_file_following_symlinks(sqfs* fs, sqfs_inode* inode, const char* target) {
-    if (!appimage_type2_resolve_symlink(fs, inode)) {
+    if (!appimage_type2_resolve_symlink(fs, inode, 4)) {
```

Comment:
Why `4`?

Reply:
Seemed enough length for a chain of symlinks to me. But if you know a use case where more than 4  symlinks are chained, we can raise that number.

Reply:
FIXED